core/src/main/java/hudson/model/queue/MappingWorksheet.java

```diff
@@ -387,7 +387,7 @@ public MappingWorksheet(BuildableItem item, List<? extends ExecutorSlot> offers,
         for (List<SubTask> group : m.values()) {
             works.add(new WorkChunk(group,works.size()));
         }
-        this.works = ImmutableList.copyOf(works);
+        this.works = Collections.unmodifiableList(works);
```

Comment:
The local variable `works` was declared within this method and goes out of scope at the end of the method, so nothing else can modify it. We do not need to copy it.